apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/purchase-notice/index.jsx

```diff
@@ -0,0 +1,34 @@
+import { useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useEffect, useState } from 'react';
+import './style.scss';
+
+function PurchaseNotice() {
+	const [ hasPaymentNotice, setHasPaymentNotice ] = useState( false );
```

Comment:
Minor, but if you use `useRef()` for this instead, it will not cause a re-render when the state is changed.

Reply:
Addressed by #66138